src/utils/embedding_store.py

```
@@ -293,6 +293,69 @@ async def _get_observations_for_reasoning(
+
+            if total_without_context + len(context_tokens) <= max_tokens:
+                return f"{query_prefix}{query}{context_prefix}{conversation_context}"
+
+            # Truncate context to fit
+            available_context_tokens = max_tokens - total_without_context
+            if available_context_tokens > 0:
+                truncated_context = encoding.decode(
+                    context_tokens[-available_context_tokens:]
+                )
+                return f"{query_prefix}{query}{context_prefix}{truncated_context}"
+            else:
```

Ruff (0.12.2)

Unnecessary `else` after `return` statement

Remove unnecessary `else`

(RET505)